### DISCUSSION TITLE
linux-arm-openssl-1.1.x  url incorrect

### DISCUSSION THREAD

**Discussion by reharik** [prisma/prisma#9656] [Q&A] [answered]
Hi, from inside a node docker container, when I run `prisma migrate dev reset` I get the following error.  

`Downloading Prisma engines for Node-API for debian-openssl-1.1.x and linux-arm-openssl-1.1.x and linux-musl [        ] 0%Error: Not Found https://binaries.prisma.sh/all_commits/afdab2f10860244038c4e32458134112852d4dad/linux-arm-openssl-1.1.x/libquery_engine.so.node.gz`

This is definitely from within prisma code. This url is wrong.  When one tries to hit it from the browser I get the following.  

`Error>
<Code>NoSuchKey</Code>
<Message>The specified key does not exist.</Message>
<Key>all_commits/afdab2f10860244038c4e32458134112852d4dad/linux-arm-openssl-1.1.x/libquery_engine.so.node.gz</Key>
<RequestId>ZHEFSRK4QHMV712A</RequestId>
<HostId>Zpo8LxrcP9uQLk2Lcmvs25MDTMq2AOpa12VKjUVY9aKjhQHZwQ2hZWD0nx8i6Jqh5MZb5iJqqVA=</HostId>
</Error>`

I suspect that I probably don't need this for running prisma inside the docker container.  Seems this works ` ["native", "debian-openssl-1.1.x",  "linux-musl"]`  but I believe that when running locally mac people need that line.  
Any help would be appreciated

Thanks,
Raif

**Comment by ryands17** [comment]
@reharik 👋 

Could you share your `Dockerfile` and Prisma version?

**Reply by janpio** [reply]
- `afdab2f10860244038c4e32458134112852d4dad` is from 3.2.0. 
- `linux-arm-openssl-1.1.x` is not something we compile, we only offer `linux-arm64-openssl-1.1.x`: https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#linux-arm64
- Question now is why did installation tried to download the old package
- And why adding some binary targets fixes something (Where exactly did you add that @reharik? What did you have before?)

Did this work in older versions @reharik?

**Reply by reharik** [reply]
Seems the arm64 works.  I guess the arm version was from before the upgrade.  
Thank you all.
R

**Reply by janpio** [reply]
Why is/was your application trying to use the `linux-arm` version? Did you configure that somewhere manually?

**Reply by reharik** [reply]
so we have the api running in a container, but some of our prisma commands are execute locally.  locally we have a bunch of macs, and me.  linux.  So we were getting errors when trying to build the client, different error per os.  So we had to add several binary targets.  In fact it looks like this now.

`generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["native", "debian-openssl-1.1.x", "linux-arm64-openssl-1.1.x", "linux-musl"]
  output        = "./generated/client"
}`

Although the arm64 one doesn't work locally.  So I had to take that out for now.  I'd love it if you'd tell me I'm doing this all wrong.  It just hard to coordinate a debug across OS remotely. 
Thanks
R

**Reply by janpio** [reply, marked as answer]
If you build the client on another platform than where you run it, using `binaryTargets` is indeed the right way to go. Usually you would develop on Mac or Windows, then deploy to Linux - and add e.g. `"native", "debian-openssl-1.1.x"` to the binaryTargets to make that work properly. `native` becomes the one of your local machine, and then via the other one it is sure that the correct deployment platform engine is also included.

**Reply by phocks** [reply]
Also getting this trying to run on Raspberry Pi. So you're saying arm7 is unsupported?

**Reply by DylanDelobel** [reply]
Same error on a Raspberry Pi

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
